src/entity_builder.rs

```diff
@@ -53,38 +53,98 @@ impl EntityBuilder {
             storage: Box::new([]),
             info: Vec::new(),
             ids: Vec::new(),
-            id_set: HashSet::new(),
+            indices: HashMap::new(),
         }
     }
 
     /// Add `component` to the entity
```

Comment:
Let's document the behavior when a component already exists for both versions.

Reply:
Added docs for this in the commit I just pushed; please take a look and see if they're adequate.